cloud/ec2.go

```diff
@@ -687,6 +689,36 @@ func (m *ec2Manager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, e
 				"host_provider": h.Distro.Provider,
 				"distro":        h.Distro.Id,
 			}))
+			if h.ShouldFallbackToOnDemand() && (strings.Contains(err.Error(), EC2InsufficientCapacity) || strings.Contains(err.Error(), EC2UnfulfillableCapacity)) {
+				// Spot creation failed due to an InsufficientCapacity error.
+				// Try to spawn the host as OnDemand instead of Spot
```

Comment:
I would remove this comment. It is unnecessary since the code is quite clear in what it is doing.

Reply:
Hmm... maybe it'd be helpful if I rephrased it? I feel like it would help future developers to know that we are doing this because Spot requests fail with these errors and that falling back to OnDemand fixes it, which I think is less clear from the code (and the current comment as is). I could also leave that to the ticket trail on git blame though. Not sure what the standard is.

Reply:
Ok I gave it a stab that rephrases it more along the "why" dimension, but happy to remove it if we prefer to keep that as a rule to tickets and git blame.